### ISSUE TITLE
Battery widget: weird time remaining output

### ISSUE THREAD

**Issue by ramnes** [qtile/qtile#1314]
Since the last updates of the battery widget, the time remaining printed on my laptop does not make much sense:

![Screenshot](https://i.imgur.com/R1vPvtN.png)

It is most likely related to those errors in my logs:

```
2019-04-26 12:24:23,222 ERROR libqtile battery.py:_load_file():L235 Failed to get current_now
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/libqtile/widget/battery.py", line 233, in _load_file
    return f.read().strip(), value_type
OSError: [Errno 19] No such device
2019-04-26 12:24:23,223 ERROR libqtile battery.py:_load_file():L235 Failed to get power_now
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/libqtile/widget/battery.py", line 232, in _load_file
    with open(path, 'r') as f:
FileNotFoundError: [Errno 2] No such file or directory: '/sys/class/power_supply/BAT0/power_now'
2019-04-26 12:24:23,223 ERROR libqtile battery.py:_load_file():L235 Failed to get current_now
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/libqtile/widget/battery.py", line 233, in _load_file
    return f.read().strip(), value_type
OSError: [Errno 19] No such device
```

**Comment by ep12** [comment]
I have very similar error messages with energy_... mostly. The battery icon has disappeared completely from my bar and I haven't found the source of this. As far as I know qtile checks two file name variants and one is existing yet I don't have a battery indicator...

**Comment by Etn40ff** [comment]
This seems to be caused by the fact that in `_LinuxBattery.update_status` time is changed after power has been rescaled. Here is a dirty fix

```
--- a/libqtile/widget/battery.py
+++ b/libqtile/widget/battery.py
@@ -284,12 +284,6 @@ class _LinuxBattery(_Battery, configurable.Configurable):
         else:
             percent = now / full
 
-        if power_unit == 'uA':
-            voltage = float(self._get_param('voltage_now_file')[0])
-            power = voltage * power / 1e12
-        elif power_unit == 'uW':
-            power = power / 1e6
-
         if power == 0:
             time = 0
         elif state == BatteryState.DISCHARGING:
@@ -297,6 +291,12 @@ class _LinuxBattery(_Battery, configurable.Configurable):
         else:
             time = int((full - now) / power)
 
+        if power_unit == 'uA':
+            voltage = float(self._get_param('voltage_now_file')[0])
+            power = voltage * power / 1e12
+        elif power_unit == 'uW':
+            power = power / 1e6
+
         return BatteryStatus(state=state, percent=percent, power=power, time=time)
```

**Comment by zordsdavini** [comment]
@Etn40ff your diff works. Can it be done as pull request?

**Comment by Etn40ff** [comment]
Done

**Comment by ramnes** [comment]
Fixed with #1373. Thanks!
